docs/wiki/get-started/More-with-CSP.md

```diff
@@ -1,141 +1,176 @@
-csp offers several built-in generic, mathematical, and statistical nodes that are often required in streaming workflows. This allows you to write applications quickly, and update and expand them as required by including new nodes.
+CSP offers several built-in generic, mathematical, and statistical nodes that are often required in streaming workflows. This allows you to write applications quickly, and update and expand them as required by including new nodes.
 
-In this tutorial, you will calculate the [volume weighted average price (VWAP)](https://www.investopedia.com/terms/v/vwap.asp) and the profit and loss (PnL) of the trade, on a stream of trades. Check out the complete example: [trade profit-and-loss example](examples/01_basics/e4_trade_pnl.py)
+In this tutorial, you build a CSP program to track a sample shopping cart and purchases, where a 10% discount if the purchase is made within 1 minute. You will learn to create and use composite data types, and add more functionality to nodes.
+
+Check out the complete example: [Retail cart example](examples/01_basics/e5_retail_cart.py).
 
 ## Compound inputs with `csp.Struct`
 
-A Trade consists of multiple parts: price of a share, quantity of shares, and indication of a buying or a selling transaction.
+A Cart consists of multiple parts: product name, quantity, indication if the product was added or removed from the cart, and if a purchase was made.
 
-`struct` is a useful data type for this type of information. In csp, you can use [`csp.Struct`](csp.Struct-API), a higher-level data type that can be defined with type-annotated values.
+A `struct` (in the C programming language) is a useful data type for this type of information. In CSP, you can use [`csp.Struct`](csp.Struct-API), a higher-level data type that can be defined with type-annotated values.
 
 ```python
-class Trade(csp.Struct):
-    price: float
+class Cart(csp.Struct):
+    product: str
     qty: int
-    buy: bool
+    add: bool
+    purchase: bool
 ```
 
 ## Build computing nodes
 
-To calculate volume-weighted averages, you need the cumulative sum of previous trade prices and quantities. Hence, your csp node needs to store stateful information. You can use \[`csp.state`\] to declare stateful variables that bound to the node.
+### Track cart updates
+
+To track the total cart items and price, you need stateful variables that can be updated each time the cart is modified.
+You can use `csp.state` to declare stateful variables that are bound to a node.
+
+> \[!TIP\]
+> By convention, state variables are prefixed with `s_`.
 
-A csp node can return multiple named outputs, denoted as `csp.Outputs` type, created using the `csp.output` function. The individual return values can be accessed with dot notation.
+A CSP node can also return multiple named outputs, denoted as `csp.Outputs` type, created using the `csp.output` function. The individual return values can then be accessed with dot notation.
```

Comment:
"created using the csp.output function" is a bit confusing / misleading.  The outputs edges arent being created, they are being returned / ticked.
Maybe we can reword as "ticked by using ...."